clang/lib/Headers/hlsl/hlsl_intrinsics.h

```diff
@@ -1295,11 +1295,13 @@ double4 trunc(double4);
 /// true, across all active lanes in the current wave.
 _HLSL_AVAILABILITY(shadermodel, 6.0)
 _HLSL_BUILTIN_ALIAS(__builtin_hlsl_wave_active_count_bits)
+__attribute__((convergent))
```

Comment:
Should remove the convergent attribute (we never should have added this). The language settings need to set IR convergent on everything

Reply:
@llvm-beanz FYI. Seems like I'll remove it then.

Reply:
I'm not sure I entirely agree with @arsenm here.

From the LLVM-IR and compiler implementation perspective it is helpful to be able to expect that convergent is the default expected behavior, but from the language perspective for HLSL it is preferable to imply that convergence is the outlier because in the common case convergence isn't required.

I really don't want to tie the implementation of HLSL to a 4.5 year old PR that isn't merged.

@bogner, do you have thoughts here?

Reply:
Convergence requires all transitive callers to also be convergent. No real code is going to do this. The existence of convergence operations means you have a convergent language and must optimize to known-not-convergent

Reply:
Right, so in that case, I'll add the convergent attribute again, and later down the road, once we have the noconvergent-default in place, we'll be able to flip this back.
Thanks all for the context and explanations!

Reply:
> Convergence requires all transitive callers to also be convergent. No real code is going to do this. The existence of convergence operations means you have a convergent language and must optimize to known-not-convergent

In languages like OpenCL and HLSL, this requirement is actually fine and we don't need it explicitly annotated. OpenCL and HLSL are effectively single-source kernel languages and all code gets inlined. So we can do robust attribute propagation.

Reply:
I filed an issue to track enabling `-fconvergent-functions` for HLSL (#86506).